lib/ae_mdw/txs.ex

```diff
@@ -159,40 +159,45 @@ defmodule AeMdw.Txs do
           add_spendtx_details?()
         ) :: {:ok, page_cursor(), [tx()], page_cursor()} | {:error, Error.t()}
   def fetch_txs(state, pagination, range, query, cursor, add_spendtx_details?) do
-    with {:ok, ids_fields} <- extract_ids_fields(query) do
-      types = query |> Map.get(:types, MapSet.new()) |> MapSet.to_list()
-      cursor = deserialize_cursor(cursor)
-      ids_fields = Enum.to_list(ids_fields)
+    with {:ok, streamer} <- txs_streamer(state, range, query, cursor) do
+      do_fetch_txs(state, streamer, pagination, add_spendtx_details?)
+    end
+  end
 
-      scope =
-        case range do
-          {:gen, first_gen..last_gen} ->
-            {DbUtil.first_gen_to_txi(state, first_gen), DbUtil.last_gen_to_txi(state, last_gen)}
+  @spec count_micro_block_txs(state(), binary()) :: {:ok, non_neg_integer()} | {:error, Error.t()}
+  def count_micro_block_txs(state, hash) do
```

Comment:
Currently the `/micro-blocks/:hash` returns the count of transactions as part of the response, is there a reason we need a separate endpoint?

Reply:
It will be used on the front-page, to avoid over-fetching until we don't have GraphQL (:

Reply:
Replaced the streaming with tx indexes math though.

Reply:
To avoid adding a new endpoint, perhaps we can use the already existing `/txs/count` endpoint, but include a new `block_hash=mh_...` filter, this way we can also start supporting `?type=spend&block_hash=mh_..` as per #1471 requirements

Reply:
The filtering would come next and it was avoiding mixing with other params as `/txs/count` accepts many but looks as a good idea. It could continue to be clear and an exception is launched when scope is used. 
PS: only mb_hash is accept since there might thousands of txs on a generation.